geopandas/vectorized.pyx

```diff
@@ -0,0 +1,1029 @@
+
+    def centroid(self):
+        return geo_unary_op('centroid', self.data)
+
+    def convex_hull(self):
+        return geo_unary_op('convex_hull', self.data)
+
+    def envelope(self):
+        return geo_unary_op('envelope', self.data)
+
+    def representative_point(self):
+        return geo_unary_op('representative_point', self.data)
+
+    def area(self):
+        return vector_float('area', self.data)
+
+    def length(self):
+        return vector_float('length', self.data)
+
+    def difference(self, other):
+        return self.binary_geo(other, 'difference')
+
+    def symmetric_difference(self, other):
+        return self.binary_geo(other, 'symmetric_difference')
+
+    def union(self, other):
+        return self.binary_geo(other, 'union')
+
+    def intersection(self, other):
+        return self.binary_geo(other, 'intersection')
+
+    def buffer(self, distance, resolution=16, cap_style=CAP_STYLE.round,
+              join_style=JOIN_STYLE.round, mitre_limit=5.0):
+        return buffer(self.data, distance, resolution, cap_style, join_style,
+                      mitre_limit)
+
+    # for Series/ndarray like compat
+
+    @property
+    def shape(self):
+        """ Shape of the ...
+
+        For internal compatibility with numpy arrays.
+
+        Returns
+        -------
+        shape : tuple
+        """
+
+        return tuple([len(self)])
+
+    def ravel(self, order='C'):
+        """ Return a flattened (numpy) array.
+
+        For internal compatibility with numpy arrays.
+
+        Returns
+        -------
+        raveled : numpy array
+        """
+        return np.array(self)
+
+    def view(self):
+        """Return a view of myself.
+
+        For internal compatibility with numpy arrays.
+
+        Returns
+        -------
+        view : Categorical
+           Returns `self`!
+        """
+        return self
+
+    def to_dense(self):
+        """Return my 'dense' representation
+
+        For internal compatibility with numpy arrays.
+
+        Returns
+        -------
+        dense : array
+        """
+        return to_shapely(self.data)
+
+    def get_values(self):
+        """ Return the values.
+
+        For internal compatibility with pandas formatting.
+
+        Returns
+        -------
+        values : numpy array
+            A numpy array of the same dtype as categorical.categories.dtype or
+            Index if datetime / periods
+        """
+        # if we are a datetime and period index, return Index to keep metadata
+        return to_shapely(self.data)
+
+    def __array__(self, dtype=None):
+        """
+        The numpy array interface.
+
+        Returns
+        -------
+        values : numpy array
+            A numpy array of either the specified dtype or,
+            if dtype==None (default), the same dtype as
+            categorical.categories.dtype
+        """
+        ret = self.data
+        if dtype:
+            return np.asarray(ret, dtype)
+        return ret
```

Comment:
Do you have a preference here? returning the pointers vs array of densified shapes ?